### DISCUSSION TITLE
Issue with rendering of @livewireScripts($nonce)

### DISCUSSION THREAD

**Discussion by sscotti** [livewire/livewire#4424] [7. Bugs (archived)]
Not sure this is really an issue, but I noticed that when the output gets rendered in my browser it appears as:

`<script src="/vendor/livewire/livewire.js?id=b09cb328e689f1bb8d77" data-turbo-eval="false" data-turbolinks-eval="false"nonce="dmj1S6GBM`

That apparently comes from:  vendor/livewire/livewire/src/LivewireManager.php

In the source that looks like:

$nonce = isset($options['nonce']) ? **"nonce**=\"{$options['nonce']}\"" : '';

and if you change to:

$nonce = isset($options['nonce']) ? **" nonce**=\"{$options['nonce']}\"" : '';

the space, it puts a space between the false and the nonce.

The reason I noticed it is because it is "red" in the view rendered source in FireFox.

/sds

**Comment by joshhanley** [comment]
@sscotti yeah that should probably be changed. Are you able to submit a PR? Thanks!

**Reply by sscotti** [reply]
Your e-mail says to show a screenshot:

![image](https://user-images.githubusercontent.com/236933/147308483-228f4bd8-c6c4-41b4-ac18-1d90da0471b0.png)
![image](https://user-images.githubusercontent.com/236933/147308449-f39c0aea-2cdc-4d9f-bb3a-63460a4770f7.png)

vs.

![image](https://user-images.githubusercontent.com/236933/147308511-8ff65848-be9f-4970-96bc-66db4036af0e.png)
![image](https://user-images.githubusercontent.com/236933/147308543-d1819085-b74a-4e0c-ae5b-2ffeddec5d31.png)

You might want to confirm it is really an issue in some other application.  I don't think it actually affects my app, bit I don't think I am actually using the nonces there anyway.  They are there in case I need them later.

I'd have to check my version and everything.  Really just a curiosity when I saw it, but could be a problem.

**Reply by sscotti** [reply]
I am behind a bit:  composer show shows:

livewire/livewire                           v2.5.3        A front-end frame...

I think 241 in your current code:        $nonce = isset($options['nonce']) ? "nonce=\"{$options['nonce']}\"" : '';

**Reply by joshhanley** [reply]
@sscotti first time I read what you'd said, I didn't understand it, but then I read again and did, hence update my commented, sorry! 😅

**Reply by joshhanley** [reply]
But yeah I think there should definitely be a space in there

**Reply by sscotti** [reply]
Now might be a good time to upgrade.  I just upgraded my composer require and get 2.8x, but now I must have to publish the assets also.  Have to think about that because I don't want to overwrite anything.

https://github.com/livewire/livewire/issues/2320

**Reply by joshhanley** [reply]
@sscotti you shouldn't need to republish everything, just the JS assets (or not publish them at all is a better option), see my post here about removing the published assets https://github.com/livewire/livewire/discussions/3769

**Reply by joshhanley** [reply]
Publishing just the assets, and not the config or pagination shouldn't impact anything. There is flags just for publishing the individual pieces rather than just doing `livewire:publish`

**Reply by sscotti** [reply]
Sorry.  Anyways.  Doesn't seem to be a problem with the current release after upgrading.  You might just want to keep that in mind.  You can probably tell from the code if there should be a space there or not.  Thanks.

Sorry I brought it up. ):

        "livewire/livewire": "2.5.3",
        "livewire/livewire": "^2.8",

**Reply by joshhanley** [reply]
> but now I must have to publish the assets also. Have to think about that because I don't want to overwrite anything.

@sscotti my above two comments were in relation to this, sorry if that wasn't clear 🙂

Don't be sorry about bring it up! Happy to work through it 🙂

**Comment by sscotti** [comment]
I’ve seen stranger.  It might have something to do with my layout.  As I recall I made some adjustments before you fully supported nonces with CSS and JS.

However, before I upgraded I got that warning, and after I did not.  The extra space won’t hurt though ?

My layout has:

       <?php $nonce = ["nonce" => csp_nonce()] ?>
        @livewireScripts($nonce)

Which apparently can be something else now ?

In my view source I now have:

And:
Which seems correct.
___________________

Stephen D. Scotti, M.D.
Some Idiots Are Savants

  E-mails:  ***@***.***,  ***@***.***
   Mobile:  +1 651 313 0209
Address:  2508 Delaware St SE, Unit 163, Minneapolis, MN 55414

  sscotti
  www.linkedin.com/in/stephen-d-scotti
  https://github.com/sscotti
  https://idiots-are-savants.com (for fun). https://www.sias.dev (for business)

___________________
The content of this email is confidential and intended for the recipient specified in message only. It is strictly forbidden to share any part of this message with any third party, without a written consent of the sender. If you received this message by mistake, please reply to this message and follow with its deletion, so that we can ensure such a mistake does not occur in the future.
